### DISCUSSION TITLE
How to add validations to Organization names

### DISCUSSION THREAD

**Discussion by carao2020** [community/community#23407] [Enterprise]
<p>I am working for an organization which is using Enterprise GitHub.<br>
We have requirement to enforce validations on the GitHub Organization name while creation of the same.<br>
For example, we would like to allow creating an organization name that should start with “ABC-”.</p>
<p>Could you please help us if there is any way that we can implement this. Thanks in advance</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>:wave: Welcome!</p>
<p>There isn’t a setting built in, but if you’re using Enterprise Server (self-hosted), you could create a rule like that using <a href="https://docs.github.com/en/enterprise-server@2.20/rest/reference/enterprise-admin#global-webhooks">global webhooks</a>.</p>
<p>This would require an application for the org creation webhooks to be sent to, which would execute the naming convention business logic. This application could take further steps as necessary, from messaging the appropriate person to using the GitHub API to change the organisation name.</p>

**Comment by carao2020** [comment]
<p>Thank you very much <a class="mention" href="/canuckjacq">@canuckjacq</a>,</p>
<p>This would really help us.</p>
<p>One more clarification regarding the validations that we can implement on the UI. Could you please suggest if there is a way that we can implement validations at the UI level.</p>
<p>For example, the existing Organization names can accept only alphanumeric characters with “-”.  Could you please suggest how these are implemented so that we can extend these validations to our requirement.</p>
<p>Additionally, kindly suggest if we can utilize GitHub APIs to rename an existing Organizations.</p>
<p>Thanks in advance</p>

**Comment by carao2020** [comment]
<aside class="quote no-group" data-username="carao2020" data-post="3" data-topic="185656">
<div class="title">
<div class="quote-controls"></div>
 carao2020:</div>
<blockquote>
<p><a class="mention" href="/canuckjacq">@canuckjacq</a>,</p>
</blockquote>
</aside>
<p>Hi <a class="mention" href="/canuckjacq">@canuckjacq</a>,</p>
<p>Could you please help to review above clarification and suggest. Thanks in advance.</p>
